### DISCUSSION TITLE
power outage problem starting esp8266

### DISCUSSION THREAD

**Discussion by Gandalf4678** [micropython/micropython#10823] [ESP8266]
Hello, 
I made a setup using an ESP8266, using micropython for development. This setup allows me to retrieve temperature data from a DS18B20 sensor and send the information to a broker topic via WiFi.

I have a problem that occurs during a power outage. I have to press the reset button for the program to restart correctly. How can I avoid this?
thank

**Comment by robert-hh** [comment]
As Gandalf you probably know some magic.
But jokes aside: That's a problem than can not be solved by software. You will need some external Reset circuitry which performs a clean reset to the board, when the power comes up again. Like a LP3470 or TL7733b.

**Reply by Gandalf4678** [reply]
thank you, i will see how to implement it
